File: bot-api/src/main/java/gg/mineral/bot/api/event/EventHandler.java
package gg.mineral.bot.api.event;

import org.eclipse.jdt.annotation.NonNull;

public interface EventHandler {
    /**
     * Schedules a task to run when an event occurs.
     * 
     * @param event
     *              the event
     * 
     * @return true if the event was cancelled
     * 
     */
    <T extends Event> boolean callEvent(@NonNull T event);
}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/event/EventHandler.kt
package gg.mineral.bot.api.event

interface EventHandler {
    /**
     * Schedules a task to run when an event occurs.
     *
     * @param event
     * the event
     *
     * @return true if the event was cancelled
     */
    fun <T : Event> callEvent(event: T): Boolean
}
